Add cisagov's standard PR checklists to the Lineage templates

jsf9k opened the pull request:
## 🗣 Description ##

This pull request adds [the cisagov org's standard PR checklists](https://github.com/cisagov/.github/blob/develop/.github/pull_request_template.md) to the Lineage templates.

## 💭 Motivation and context ##

This change was triggered by the realization that one will likely need to bump the version and create a release after merging a Lineage PR for any versioned repositories.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

jsf9k committed:
Add cisagov's standard PR checklists to the Lineage templates

This was triggered by the realization that one will likely need to
bump the version and create a release after merging a Lineage PR for
any versioned repositories.

jsf9k committed:
Remove conflict resolution checkbox from clean template

jsf9k committed:
Bump version from 0.2.0 to 0.2.1

dav3r commented on src/lineage/templates/conflict_template.md:
```diff
@@ -50,12 +50,44 @@ that you must resolve before merging this pull request!
-1. Check the "Everything is cool" checkbox below:
-
-    - [ ] ✌️ The conflicts in this pull request have been resolved.
+1. Check the "Everything is cool" checkbox below.
```
These directions should be updated to something like:
```suggestion
1. Confirm each item in the "Pre-approval checklist" below.

1. Remove any of the checklist items that do not apply.

1. Ensure every remaining checkbox has been checked.
```

dav3r commented on src/lineage/templates/clean_template.md:
```diff
@@ -9,6 +9,39 @@ Upstream repository: [`{{ remote_url }}`]({{ remote_url }})
 
+## ✅ Pre-approval checklist ##
+
+<!-- Remove any of the following that do not apply. -->
+<!-- Draft PRs should have one or more unchecked boxes. -->
+<!-- If you're unsure about any of these, don't hesitate to ask. -->
+<!-- We're here to help! -->
```
Say the quiet part out loud, since these are Lineage PRs:
```suggestion
- Remove any of the following that do not apply.
- If you're unsure about any of these, don't hesitate to ask. We're here to help!
```

dav3r commented on src/lineage/templates/clean_template.md:
```diff
@@ -9,6 +9,39 @@ Upstream repository: [`{{ remote_url }}`]({{ remote_url }})
+
+## ✅ Pre-merge checklist ##
+
+<!-- Remove any of the following that do not apply. -->
+<!-- These boxes should remain unchecked until the pull request has been -->
+<!-- approved. -->
```
```suggestion
- Remove any of the following that do not apply.
- These boxes should remain unchecked until the pull request has been approved.
```

dav3r commented on src/lineage/templates/clean_template.md:
```diff
@@ -9,6 +9,39 @@ Upstream repository: [`{{ remote_url }}`]({{ remote_url }})
+
+## ✅ Post-merge checklist ##
+
+<!-- Remove any of the following that do not apply. -->
```
```suggestion
- Remove any of the following that do not apply.
```

dav3r commented on src/lineage/templates/conflict_template.md:
```diff
@@ -50,12 +50,44 @@ that you must resolve before merging this pull request!
 
+## ✅ Pre-approval checklist ##
+
+<!-- Remove any of the following that do not apply. -->
+<!-- Draft PRs should have one or more unchecked boxes. -->
+<!-- If you're unsure about any of these, don't hesitate to ask. -->
+<!-- We're here to help! -->
```
```suggestion
- Remove any of the following that do not apply.
- If you're unsure about any of these, don't hesitate to ask. We're here to help!
```

dav3r commented on src/lineage/templates/conflict_template.md:
```diff
@@ -50,12 +50,44 @@ that you must resolve before merging this pull request!
+
+## ✅ Pre-merge checklist ##
+
+<!-- Remove any of the following that do not apply. -->
+<!-- These boxes should remain unchecked until the pull request has been -->
+<!-- approved. -->
```
```suggestion
- Remove any of the following that do not apply.
- These boxes should remain unchecked until the pull request has been approved.
```

dav3r commented on src/lineage/templates/conflict_template.md:
```diff
@@ -50,12 +50,44 @@ that you must resolve before merging this pull request!
+
+## ✅ Post-merge checklist ##
+
+<!-- Remove any of the following that do not apply. -->
```
```suggestion
- Remove any of the following that do not apply.
```

dav3r reviewed (changes requested):
I've got some suggestions.

jsf9k committed:
Elaborate on checkbox portion of instructions

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Move helper statements out of comments and into the light

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
👍 ✅

mcdonnnj commented:
@jsf9k Would you consider this PR as resolving #61? If so we should link it to this PR.

jsf9k commented:
> @jsf9k Would you consider this PR as resolving #61? If so we should link it to this PR.

Done!

mcdonnnj commented on src/lineage/templates/clean_template.md:
```diff
@@ -9,6 +9,36 @@ Upstream repository: [`{{ remote_url }}`]({{ remote_url }})
+- [ ] All relevant repo and/or project documentation has been updated
+      to reflect the changes in this PR.
+- [ ] Tests have been added and/or modified to cover the changes in this PR.
+- [ ] All new and existing tests pass.
```
Since this template is for an automatically generated pull request I do not think all of these bullets are necessarily appropriate.

```suggestion
- [ ] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [ ] All relevant type-of-change labels have been added.
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.
```

mcdonnnj commented on src/lineage/templates/conflict_template.md:
```diff
@@ -50,12 +50,45 @@ that you must resolve before merging this pull request!
+- If you're unsure about any of these, don't hesitate to ask. We're here to help!
+
+- [ ] ✌️ The conflicts in this pull request have been resolved.
+- [ ] Changes are limited to a single goal - *eschew scope creep!*
+- [ ] *All* future TODOs are captured in issues, which are referenced
+      in code comments.
+- [ ] All relevant type-of-change labels have been added.
+- [ ] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
+- [ ] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
+- [ ] All relevant repo and/or project documentation has been updated
+      to reflect the changes in this PR.
+- [ ] Tests have been added and/or modified to cover the changes in this PR.
+- [ ] All new and existing tests pass.
```
Same thought about item relevancy.

```suggestion
- [ ] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [ ] All relevant type-of-change labels have been added.
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.
```

mcdonnnj commented on src/lineage/templates/clean_template.md:
```diff
@@ -9,6 +9,36 @@ Upstream repository: [`{{ remote_url }}`]({{ remote_url }})
+- Remove any of the following that do not apply.
+- These boxes should remain unchecked until the pull request has been approved.
+
+- [ ] Bump version via the `bump_version.sh` script, if this
+      repository is versioned.
```
I would like to clarify in this point that this should only be done if it is appropriate. Following semver not every group of Lineage changes should see a version change (GitHub Actions changes, pre-commit hook updates, etc.).

jsf9k replied:
Please see commit 3364e18.

mcdonnnj commented on src/lineage/templates/conflict_template.md:
```diff
@@ -50,12 +50,45 @@ that you must resolve before merging this pull request!
+- Remove any of the following that do not apply.
+- These boxes should remain unchecked until the pull request has been approved.
+
+- [ ] Bump version via the `bump_version.sh` script, if this
+      repository is versioned.
```
Same thought about version bumping.

jsf9k replied:
Please see commit 3364e18.

jsf9k committed:
Remove checklist items that do not apply to automatically generated PRs

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Clarify that a version bump should only be done when appropriate

Changes to internal dependencies that do not change the API (or the
build product) do not require and should not receive a version bump.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj reviewed (approved):
LGTM ✔
